iron-fit-behavior.html

```diff
@@ -362,25 +368,19 @@
       var left = position.left + margin.left;
       var top = position.top + margin.top;
 
-      // Use original size (without margin).
-      var right = Math.min(fitRect.right - margin.right, left + rect.width);
-      var bottom = Math.min(fitRect.bottom - margin.bottom, top + rect.height);
+      // We first limit right/bottom, then use those values to limit top/left.
+      var maxRight = fitRect.right - margin.right - this.horizontalOffset * (position.horizontalAlign !== 'left');
+      var maxBottom = fitRect.bottom - margin.bottom - this.verticalOffset * (position.verticalAlign !== 'top');
+      var right = Math.min(maxRight, left + rect.width);
+      var bottom = Math.min(maxBottom, top + rect.height);
 
-      var minWidth = this._fitInfo.sizedBy.minWidth;
-      var minHeight = this._fitInfo.sizedBy.minHeight;
-      if (left < margin.left) {
-        left = margin.left;
-        if (right - left < minWidth) {
-          left = right - minWidth;
-        }
-      }
-      if (top < margin.top) {
-        top = margin.top;
-        if (bottom - top < minHeight) {
-          top = bottom - minHeight;
-        }
-      }
+      // Keep left/top within fitInto and respect minWidth/minHeight.
+      var minLeft = fitRect.left + margin.left + this.horizontalOffset * (position.horizontalAlign === 'left');
+      var minTop = fitRect.top + margin.top + this.verticalOffset * (position.verticalAlign === 'top');
+      left = Math.min(Math.max(left, minLeft), right - this._fitInfo.sizedBy.minWidth);
+      top = Math.min(Math.max(top, minTop), bottom - this._fitInfo.sizedBy.minHeight);
```

Comment:
I've noticed that `bottom - this._fitInfo.sizedBy.minHeight` can be less than `minTop` (similarly for the stuff in `left`), so this clamp can end up causing `top` to be less than `minTop`. I think reversing the order of the `min`/`max` calls here and changing `(bottom - top)` to `Math.max(_fitInfo...minHeight, bottom - top)` below might fix this?